arkouda/pdarrayclass.py

```diff
@@ -1452,15 +1465,18 @@ def maxk(pda : 'pdarray', k : int) -> 'pdarray':
     >>> ak.maxk(A, 3)
     array([7, 9, 10])
     """
+    if not isinstance(k, int):
+        raise TypeError('k must be an integer with a value of 1 or greater')
+    if k < 1:
+        raise ValueError('k must be 1 or greater')
+
     if isinstance(pda, pdarray):
-        if k == 0:
-            return []
         if pda.size == 0:
-            raise TypeError("must be a non-empty pdarray {} of type int or float".format(pda))
+            raise ValueError("must be a non-empty pdarray {} of type int or float".format(pda))
         repMsg = generic_msg("maxk {} {} {}".format(pda.name, k, False))
         return create_pdarray(repMsg)
     else:
-        raise TypeError("must be pdarray {}".format(pda))
+        raise TypeError("must be pdarray, not {}".format(type(pda)))
```

Comment:
`format(pda.__class__.__name__)` ?

Reply:
yes indeed, will fix. This is an example of why I'd like to work on [540](https://github.com/mhmerrill/arkouda/issues/540)